subsys/bluetooth/controller/ll_sw/ull_sched.c

```diff
@@ -4,43 +4,536 @@
+				ticker_job_sched(TICKER_INSTANCE_ID_CTLR,
+						 user_id);
+			}
+		}
+
+		LL_ASSERT(ret_cb == TICKER_STATUS_SUCCESS);
+
+		if (ticker_id == 0xff) {
+			break;
+		}
+
+		if ((ticker_id < TICKER_ID_CONN_BASE) ||
+		    (ticker_id > TICKER_ID_CONN_LAST)) {
+			continue;
+		}
+
+		conn = ll_conn_get(ticker_id - TICKER_ID_CONN_BASE);
+		if (conn && !conn->lll.role) {
+			u32_t ticks_to_expire_normal = ticks_to_expire;
+			u32_t ticks_slot_abs_curr = 0;
+#if defined(CONFIG_BT_CTLR_LOW_LAT)
+#if defined(CONFIG_BT_CTLR_XTAL_ADVANCED)
+			if (conn->evt.ticks_xtal_to_start & XON_BITMASK) {
+				u32_t ticks_prepare_to_start =
+					MAX(conn->evt.ticks_active_to_start,
+					    conn->evt.ticks_preempt_to_start);
+
+				ticks_slot_abs_curr =
+					conn->evt.ticks_xtal_to_start &
+					~XON_BITMASK;
+				ticks_to_expire_normal -=
+					ticks_slot_abs_curr -
+					ticks_prepare_to_start;
+			} else
+#endif /* CONFIG_BT_CTLR_XTAL_ADVANCED */
+			{
+				u32_t ticks_prepare_to_start =
+					MAX(conn->evt.ticks_active_to_start,
+					    conn->evt.ticks_xtal_to_start);
+
+				ticks_slot_abs_curr = ticks_prepare_to_start;
+			}
+#endif
+
+			ticks_slot_abs_curr +=
+				conn->evt.ticks_slot +
+				HAL_TICKER_US_TO_TICKS(EVENT_JITTER_US << 3);
+
+			if ((ticker_id_prev != 0xFF) &&
+			    (ticker_ticks_diff_get(ticks_to_expire_normal,
+						   ticks_to_expire_prev) >
+			     (ticks_slot_abs_prev + ticks_slot_abs))) {
+				break;
+			}
+
+			ticker_id_prev = ticker_id;
+			ticks_to_expire_prev = ticks_to_expire_normal;
+			ticks_slot_abs_prev = ticks_slot_abs_curr;
+		}
+	}
+
+	if (ticker_id_prev != 0xff) {
+		*us_offset = HAL_TICKER_TICKS_TO_US(ticks_to_expire_prev +
+						    ticks_slot_abs_prev) +
+						    (EVENT_JITTER_US << 3);
+	}
 }
 
 void ull_sched_mfy_after_mstr_offset_get(void *param)
 {
-	struct ll_scan_set *scan = param;
+	struct lll_prepare_param *p = param;
+	struct lll_scan *lll = p->param;
+	struct evt_hdr *conn_evt = HDR_LLL2EVT(lll->conn);
+	u32_t ticks_slot_overhead;
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_LOW_LAT)) {
+		ticks_slot_overhead = MAX(conn_evt->ticks_active_to_start,
+					  conn_evt->ticks_xtal_to_start);
+	} else {
+		ticks_slot_overhead = 0U;
+	}
 
-	/* TODO: */
-	scan->lll.conn_win_offset_us = 0;
+	after_mstr_offset_get(lll->conn->interval, conn_evt->ticks_slot,
+			      p->ticks_at_expire, &lll->conn_win_offset_us);
 }
 
-void ull_sched_mfy_free_win_offset_calc(void *param)
+void ull_sched_mfy_win_offset_use(void *param)
 {
-	/* TODO: */
+	struct ll_conn *conn = param;
+	u32_t ticks_slot_overhead;
+	u16_t win_offset;
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_LOW_LAT)) {
+		ticks_slot_overhead = MAX(conn->evt.ticks_active_to_start,
+					  conn->evt.ticks_xtal_to_start);
+	} else {
+		ticks_slot_overhead = 0U;
+	}
+
+	after_mstr_offset_get(conn->lll.interval, conn->evt.ticks_slot,
+				conn->llcp.conn_upd.ticks_anchor,
+				&conn->llcp_cu.win_offset_us);
+
+	win_offset = conn->llcp_cu.win_offset_us / 1250;
```

Comment:
use a #define instead of 1250; this number is also used other places in the code

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/23314